### DISCUSSION TITLE
Speech_final

### DISCUSSION THREAD

**Discussion by gdelco** [deepgram/community#453] [General help]
I have been using the speech_final flag to determine when I can handle the final response. However, sometimes the data returns a Speech_Final = False and I need to talk another time to get the final result.
Am I missing something?

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by DamienDeepgram** [comment]
You could try setting no_delay=true to avoid waiting for numbers when the last word is a number

**Comment by nikolawhallon** [comment]
I'm working on some docs to explain the various cases where speech_final will fail, and what can be done about it. Here is a preview, maybe it's helpful? https://github.com/nikolawhallon/temp-utterance-end
